pkg/controller/appsync/appsync.go

```diff
@@ -42,6 +47,20 @@ func (c *Controller) HandleSync() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
 
+		if ok, err := c.shouldSync(ctx); err != nil {
```

Comment:
Nit: this could be easier to read without the `else`

ok, err := c.shoudlSync()
...

if !ok {
  429
}